### ISSUE TITLE
[Example] Add summarization example with finetuning BART model

### ISSUE THREAD

**Issue by sxjscience** [dmlc/gluon-nlp#1484]
## Description
Currently, we still lack a summarization example and will need to add the dataset and example.

@leezu @szha

**Comment by szha** [comment]
Agreed. There's a large body of work in that space, both in extractive and nowadays abstractive summarization. I think we should survey and pick. Some good comprehensive pools to choose from: http://nlpprogress.com/english/summarization.html, https://paperswithcode.com/task/text-summarization

**Comment by sxjscience** [comment]
Actually, can we spare some efforts in adding an example this month?
